src/ranch.erl

```diff
@@ -71,7 +71,7 @@ start_listener(Ref, Transport, TransOpts0, Protocol, ProtoOpts)
 		when is_atom(Transport), is_atom(Protocol) ->
 	TransOpts = normalize_opts(TransOpts0),
 	_ = code:ensure_loaded(Transport),
-	case erlang:function_exported(Transport, name, 0) of
+	case validate_transopts(TransOpts) andalso erlang:function_exported(Transport, name, 0) of
```

Comment:
We can't return `{error, badarg}` for everything, maybe `{error, {badoption, Option}}`. Or even `{error, {badoption, Option}, 'Atom describing the error/valid values.'}`.

Reply:
In that case the return type probably needs being expanded.

Reply:
I went for the first (`error, {badoption, Option}}` with my last commit. The valid values are supposed to be in the documentation, I think.